authentication/src/main/java/io/camunda/authentication/CamundaOAuthPrincipalService.java

```diff
@@ -103,7 +105,8 @@ public OAuthContext loadOAuthContext(final Map<String, Object> claims)
     final var clientId = principals.clientId();
 
     if (username == null && clientId == null) {
-      throw new IllegalArgumentException(
+      throw new OAuth2AuthenticationException(
```

Comment:
:question: Is this returned as an error to the user instead of being logged on the server side as an error now?

Reply:
I'm trying to test that right now :eyes: